test-integration/helpers/README.md

```diff
@@ -0,0 +1,246 @@
+
+  ```coffee
+  COMMON_ELEMENTS =
+    forecastLink:
+      linkText: 'Performance Forecast'
+    studentScoresLink:
+      linkText: 'Student Scores'
+    addToggle:
+      css: '.add-assignment .dropdown-toggle'
+    planByTitle: (title) ->
+      css: ".plan label[data-title='#{title}']"
+  ```
+
+3. Create a new helper class that extends `TestHelper` and provide default options to the base constructor
+
+  ```coffee
+  class CalendarHelper extends TestHelper
+    constructor: (testContext, testElementLocator) ->
+
+      # the locator of the base UI
+      testElementLocator ?=
+        css: '.calendar-container'
+
+      # Options for this helper.
+      # The calendar has a custom loading indicator (not `.is-loading`),
+      # and often takes longer than the default wait to load time of 1000ms.
+      calendarOptions =
+        loadingLocator:
+          css: '.calendar-loading'
+        defaultWaitTime: 3000
+
+      # instantiate with options by calling the base `TestHelper`'s constructor
+      super(testContext, testElementLocator, COMMON_ELEMENTS, calendarOptions)
+
+      # Not commonly needed.
+      # For this particular UI, a planPopup and methods controlling the popup
+      # are helpful to separate from the main parent calendar UI helper methods.
+      # PlanPopupHelper is an extended TestHelper like CalendarHelper is.
+      @setCommonHelper('planPopup', new PlanPopupHelper(@test))
+  ```
+  You can read more about what happens when you instantiate [here](#helper-details).
+
+4. Extend the helper with additional UI process/action methods as needed, using el's element accessor methods:
+
+  ```coffee
+  class CalendarHelper extends TestHelper
+    constructor: ...
+    createNew: (type) =>
+      @waitUntilLoaded()
+
+      @el.addToggle.get().click()
+      ...
+
+    goPerformanceForecast: =>
+      @test.utils.windowPosition.scrollTop()
+      @el.forecastLink.get().click()
+
+    goOpen: (title) =>
+      # wait until the calendar is open
+      @waitUntilLoaded()
+      # TODO: Make this a `data-title` attribute
+      # HACK: Might need to scroll the item to click on into view
+      el = @el.planByTitle.get(title)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
+      @test.utils.windowPosition.scrollTop()
+  ```
+
+# Writing a spec
+
+With the UI helper, a spec can look like this:
+
+```coffee
+{describe, User, CourseSelect, Calendar, ReadingBuilder} = require './helpers'
+{expect} = require 'chai'
+
+TEACHER_USERNAME = 'teacher01'
+
+{CalendarHelper} = Calendar
+
+describe 'Calendar and Stats', ->
+
+  beforeEach ->
+    @user = new User(@)
+    @calendar = new CalendarHelper(@)
+    @courseSelect = new CourseSelect(@)
+
+    @user.login(TEACHER_USERNAME)
+
+  @it 'Shows stats for all published plans (readonly)', ->
+    COURSE_CATEGORY = 'BIOLOGY'
+    @courseSelect.goTo(COURSE_CATEGORY)
+    @calendar.waitUntilLoaded()
```

Comment:
Any reason `.goTo` does not already wait until loaded? It seems like adding `.waitUntilLoaded()` everywhere might get forgotten?